crates/ty/src/lib.rs

```diff
@@ -302,10 +305,12 @@ impl MainLoop {
                             return Ok(ExitStatus::Success);
                         }
 
-                        let mut stdout = stdout().lock();
-
                         if result.is_empty() {
-                            writeln!(stdout, "{}", "All checks passed!".green().bold())?;
+                            writeln!(
+                                self.printer.stdout(),
```

Comment:
Shouldn't this be printed according to the documentation of quiet:

> Silences output except for summary information.

Reply:
Fair, but I think the documentation should be changed there. I think a "success" message is not important enough to display under `--quiet`.

Reply:
How is it different from *Found x diagnostics*. It captures the same information when x is 0

Reply:
Fair enough. Maybe `stream_for_summary` isn't such a good name ;)

It might be good to add some documentation to `stdout_important` for when it should be used

Reply:
Because a message on success contains no information that isn't represented by the exit code, but on failure the message captures the number of errors?